test/utils/SCRSValidatorsSpec.scala

```diff
@@ -18,217 +18,122 @@ package utils
-      boundForm.errors.map(_.message) mustBe List("page.reg.accountingDates.date.invalid-year")
-    }
-  }
-
-  "Submitting a Accounting Dates form with invalid month field" should {
-    "return an error message if month field is invalid" in new Setup {
-      val data: Map[String, String] = Map(
-        "businessStartDate" -> "futureDate",
-        "futureDate.Year" -> "2017",
-        "futureDate.Month" -> "21",
-        "futureDate.Day" -> "1")
-      val boundForm = testAccDatesForm.bind(data)
-      boundForm.errors.map(_.message) mustBe List("page.reg.accountingDates.date.invalid-month")
-    }
-
-    "return an error message if the date is not at least two days in the future" in new Setup {
-      val data: Map[String, String] = Map(
-        "businessStartDate" -> "futureDate",
-        "futureDate.Year" -> (LocalDate.now().getYear - 1).toString,
-        "futureDate.Month" -> "12",
-        "futureDate.Day" -> "23")
-
-      val boundForm = testAccDatesForm.bind(data)
-      boundForm.errors.map(err => (err.args.head, err.message)) mustBe List(("futureDate.Day", "page.reg.accountingDates.date.future"))
-    }
-
-    "return an error message if the date is more than three years in the future" in new Setup {
-      val data: Map[String, String] = Map(
-        "businessStartDate" -> "futureDate",
-        "futureDate.Year" -> (LocalDate.now().getYear + 4).toString,
-        "futureDate.Month" -> "12",
-        "futureDate.Day" -> "23")
-
-      val boundForm = testAccDatesForm.bind(data)
-      boundForm.errors.map(err => (err.args.head, err.message)) mustBe List(("futureDate.Day", "page.reg.accountingDates.date.future"))
-    }
-
-    "return an error message if the date over 3 years in the future when today's date is 29th Feb" in new Setup(LocalDate.of(2020, 2, 29)) {
-      val data: Map[String, String] = Map(
-        "businessStartDate" -> "futureDate",
-        "futureDate.Year" -> "2023",
-        "futureDate.Month" -> "03",
-        "futureDate.Day" -> "1")
-
-      val boundForm = testAccDatesForm.bind(data)
-      boundForm.errors.map(err => (err.args.head, err.message)) mustBe List(("futureDate.Day","page.reg.accountingDates.date.future"))
+  private val errorCases = Table(
+    ("testErrorDescription", "year", "month", "day", "expectedErrors"),
+    ("all date fields are empty", "", "", "", List("page.reg.accountingDates.date.notFound")),
+    ("day field is empty", "2017", "11", "", List("page.reg.accountingDates.day.notFound")),
+    ("month field is empty", "2017", "", "11", List("page.reg.accountingDates.month.notFound")),
+    ("year field is empty", "", "11", "11", List("page.reg.accountingDates.year.notFound")),
+    ("date is invalid", "2017", "2", "31", List("page.reg.accountingDates.date.invalid-date")),
+    (
+      "any date fields are invalid",
+      "-0001",
+      "32",
+      "34",
+      List("page.reg.accountingDates.date.invalid-day", "page.reg.accountingDates.date.invalid-month")),
```

Comment:
Form validation only checks that year is an Int so doesn't return an invalid-year error for invalid year. Is covered by other date boundaries.